**1 PREPARATIONS DES FICHIERS**

**1.1 Importation des librairies**

In [3]:
#Importation de la librairie Pandas
import pandas as pd
from functools import reduce

#Importation de la librairie Numpy
import numpy as np

import pycountry
import unidecode
from fuzzywuzzy import process

**1.2 Importations des fichiers**

In [5]:
#Importation du fichier FAOSTAT_politique_2022.CSV
df_pol = pd.read_csv('FAOSTAT_politique_2022.csv')

#Importation du fichier FAOSTAT_pib_2022.CSV
df_pib = pd.read_csv('FAOSTAT_pib_2022.csv')

#Importation du fichier FAOSTAT_dispoalimentaire_2022.CSV
df_disp = pd.read_csv('FAOSTAT_dispoalimentaire_2022.csv')

#Importation du fichier Fibl_Organic_2023.CSV
df_org = pd.read_csv('Fibl_Organic_2023.csv',skiprows=7)

#Importation du fichier FAOSTAT_population_2018_2023.CSV
df_pop = pd.read_csv('FAOSTAT_population_2018_2023.csv')

#Importation du fichier FAOSTAT_investissement_2022.csv
df_inv = pd.read_csv('FAOSTAT_investissement_2022.csv')

**2.1 Analyse du fichier FAOSTAT_politique**

In [7]:
#Afficher les dimensions du fichier
print("Le tableau comporte {} observation(s) ou article(s)".format(df_pol.shape[0]))
print("Le tableau comporte {} colonne(s)".format(df_pol.shape[1]))

#La nature des données dans chacune des colonnes
print(df_pol.dtypes)

Le tableau comporte 196 observation(s) ou article(s)
Le tableau comporte 15 colonne(s)
Code Domaine               object
Domaine                    object
Code zone (M49)             int64
Zone                       object
Code Élément                int64
Élément                    object
Code Produit                int64
Produit                    object
Code année                  int64
Année                       int64
Unité                     float64
Valeur                    float64
Symbole                    object
Description du Symbole     object
Note                      float64
dtype: object


In [8]:
#Affichage les 5 premières lignes de la table
df_pol.head(5)

Code Domaine                             Domaine  Code zone (M49)  \
0           FS  Données de la sécurité alimentaire                4   
1           FS  Données de la sécurité alimentaire              710   
2           FS  Données de la sécurité alimentaire                8   
3           FS  Données de la sécurité alimentaire               12   
4           FS  Données de la sécurité alimentaire              276   

             Zone  Code Élément Élément  Code Produit  \
0     Afghanistan          6125  Valeur         21032   
1  Afrique du Sud          6125  Valeur         21032   
2         Albanie          6125  Valeur         21032   
3         Algérie          6125  Valeur         21032   
4       Allemagne          6125  Valeur         21032   

                                             Produit  Code année  Année  \
0  Stabilité politique et absence de violence/ter...        2022   2022   
1  Stabilité politique et absence de violence/ter...        2022   2022   
2  Stabilité politique et absence de violence/ter...        2022   2022   
3  Stabilité politique et absence de violence/ter...        2022   2022   
4  Stabilité politique et absence de violence/ter...        2022   2022   

   Unité  Valeur Symbole             Description du Symbole  Note  
0    NaN   -2.55       X  Ciffre de sources internationales   NaN  
1    NaN   -0.72       X  Ciffre de sources internationales   NaN  
2    NaN    0.11       X  Ciffre de sources internationales   NaN  
3    NaN   -0.74       X  Ciffre de sources internationales   NaN  
4    NaN    0.61       X  Ciffre de sources internationales   NaN

In [9]:
#Suprression des colonnes inutiles
df_pol.drop(columns=['Code Domaine', 'Domaine', 'Code Élément', 'Élément','Code année','Unité', 'Symbole', 'Description du Symbole', 'Note','Produit', 'Code Produit', 'Année'], inplace=True)
df_pol.head(5)

Code zone (M49)            Zone  Valeur
0                4     Afghanistan   -2.55
1              710  Afrique du Sud   -0.72
2                8         Albanie    0.11
3               12         Algérie   -0.74
4              276       Allemagne    0.61

In [10]:
#Renomme la colonne Valeur
df_pol = df_pol.rename(columns={"Valeur": "Stabilité politique"})
df_pol.head(5)

Code zone (M49)            Zone  Stabilité politique
0                4     Afghanistan                -2.55
1              710  Afrique du Sud                -0.72
2                8         Albanie                 0.11
3               12         Algérie                -0.74
4              276       Allemagne                 0.61

**2.2 Analyse du fichier FAOSTAT_pib_2022**

In [12]:
#Afficher les dimensions du fichier
print("Le tableau comporte {} observation(s) ou article(s)".format(df_pib.shape[0]))
print("Le tableau comporte {} colonne(s)".format(df_pib.shape[1]))

#La nature des données dans chacune des colonnes
print(df_pib.dtypes)

Le tableau comporte 183 observation(s) ou article(s)
Le tableau comporte 15 colonne(s)
Code Domaine               object
Domaine                    object
Code zone (M49)             int64
Zone                       object
Code Élément                int64
Élément                    object
Code Produit                int64
Produit                    object
Code année                  int64
Année                       int64
Unité                      object
Valeur                    float64
Symbole                    object
Description du Symbole     object
Note                      float64
dtype: object


In [13]:
#Affichage les 5 premières lignes de la table
df_pib.head(5)

Code Domaine                             Domaine  Code zone (M49)  \
0           FS  Données de la sécurité alimentaire              710   
1           FS  Données de la sécurité alimentaire                8   
2           FS  Données de la sécurité alimentaire               12   
3           FS  Données de la sécurité alimentaire              276   
4           FS  Données de la sécurité alimentaire               20   

             Zone  Code Élément Élément  Code Produit  \
0  Afrique du Sud          6126  Valeur         22013   
1         Albanie          6126  Valeur         22013   
2         Algérie          6126  Valeur         22013   
3       Allemagne          6126  Valeur         22013   
4         Andorre          6126  Valeur         22013   

                                             Produit  Code année  Année  \
0  PIB par habitant, ($ PPA internationaux consta...        2022   2022   
1  PIB par habitant, ($ PPA internationaux consta...        2022   2022   
2  PIB par habitant, ($ PPA internationaux consta...        2022   2022   
3  PIB par habitant, ($ PPA internationaux consta...        2022   2022   
4  PIB par habitant, ($ PPA internationaux consta...        2022   2022   

      Unité   Valeur Symbole             Description du Symbole  Note  
0  Int$/cap  14322.3       X  Ciffre de sources internationales   NaN  
1  Int$/cap  17261.0       X  Ciffre de sources internationales   NaN  
2  Int$/cap  13090.9       X  Ciffre de sources internationales   NaN  
3  Int$/cap  62605.5       X  Ciffre de sources internationales   NaN  
4  Int$/cap  63379.0       X  Ciffre de sources internationales   NaN

In [14]:
#Suprression des colonnes inutiles
df_pib.drop(columns=['Code Domaine', 'Domaine', 'Code Élément', 'Élément','Code année', 'Symbole', 'Description du Symbole', 'Note','Produit', 'Code Produit', 'Unité','Année'], inplace=True)
df_pib.head(5)

Code zone (M49)            Zone   Valeur
0              710  Afrique du Sud  14322.3
1                8         Albanie  17261.0
2               12         Algérie  13090.9
3              276       Allemagne  62605.5
4               20         Andorre  63379.0

In [15]:
#Renomme la colonne Valeur
df_pib = df_pib.rename(columns={"Valeur": "PIB(Int$/cap)"})
df_pib.head(5)

Code zone (M49)            Zone  PIB(Int$/cap)
0              710  Afrique du Sud        14322.3
1                8         Albanie        17261.0
2               12         Algérie        13090.9
3              276       Allemagne        62605.5
4               20         Andorre        63379.0

**2.3 Analyse du fichier FAOSTAT_dispoalimentaire_2022.CSV**

In [17]:
#Afficher les dimensions du fichier
print("Le tableau comporte {} observation(s) ou article(s)".format(df_disp.shape[0]))
print("Le tableau comporte {} colonne(s)".format(df_disp.shape[1]))

#La nature des données dans chacune des colonnes
print(df_disp.dtypes)

Le tableau comporte 3178 observation(s) ou article(s)
Le tableau comporte 15 colonne(s)
Code Domaine               object
Domaine                    object
Code zone (M49)             int64
Zone                       object
Code Élément                int64
Élément                    object
Code Produit (FBS)         object
Produit                    object
Code année                  int64
Année                       int64
Unité                      object
Valeur                    float64
Symbole                    object
Description du Symbole     object
Note                      float64
dtype: object


In [18]:
#Affichage les 5 premières lignes de la table
df_disp.head(5)

Code Domaine                      Domaine  Code zone (M49)         Zone  \
0          FBS  Bilans Alimentaires (2010-)                4  Afghanistan   
1          FBS  Bilans Alimentaires (2010-)                4  Afghanistan   
2          FBS  Bilans Alimentaires (2010-)                4  Afghanistan   
3          FBS  Bilans Alimentaires (2010-)                4  Afghanistan   
4          FBS  Bilans Alimentaires (2010-)                4  Afghanistan   

   Code Élément                                            Élément  \
0          5611                            Importations - quantité   
1          5301                           Disponibilité intérieure   
2          5521                              Aliments pour animaux   
3          5142                                         Nourriture   
4           645  Disponibilité alimentaire en quantité (kg/pers...   

  Code Produit (FBS)           Produit  Code année  Année          Unité  \
0              S2520  Céréales, autres        2022   2022         1000 t   
1              S2520  Céréales, autres        2022   2022         1000 t   
2              S2520  Céréales, autres        2022   2022         1000 t   
3              S2520  Céréales, autres        2022   2022         1000 t   
4              S2520  Céréales, autres        2022   2022  kcal/personne   

   Valeur Symbole Description du Symbole  Note  
0   30.00       I         Valeur imputée   NaN  
1   30.00       I         Valeur imputée   NaN  
2    7.00       I         Valeur imputée   NaN  
3   23.00       I         Valeur imputée   NaN  
4    0.57       E         Valeur estimée   NaN

In [19]:
#Suprression des colonnes inutiles
df_disp.drop(columns=['Code Domaine', 'Domaine', 'Code Élément','Code année', 'Symbole', 'Description du Symbole', 'Note', 'Code Produit (FBS)','Année'], inplace=True)
df_disp.head(5)

Code zone (M49)         Zone  \
0                4  Afghanistan   
1                4  Afghanistan   
2                4  Afghanistan   
3                4  Afghanistan   
4                4  Afghanistan   

                                             Élément           Produit  \
0                            Importations - quantité  Céréales, autres   
1                           Disponibilité intérieure  Céréales, autres   
2                              Aliments pour animaux  Céréales, autres   
3                                         Nourriture  Céréales, autres   
4  Disponibilité alimentaire en quantité (kg/pers...  Céréales, autres   

           Unité  Valeur  
0         1000 t   30.00  
1         1000 t   30.00  
2         1000 t    7.00  
3         1000 t   23.00  
4  kcal/personne    0.57

In [20]:
#Valeur unique Produit
valeurs_uniques = df_disp['Produit'].unique()
print("Valeurs uniques dans la colonne 'Produit' :", valeurs_uniques)

Valeurs uniques dans la colonne 'Produit' : ['Céréales, autres' 'Viande de Volailles' 'Oeufs']


In [21]:
#Récupération information céréale
df_céréale = df_disp[df_disp['Produit'].str.contains('céréale', case=False, na=False)]
df_céréale.head(5)

Code zone (M49)         Zone  \
0                4  Afghanistan   
1                4  Afghanistan   
2                4  Afghanistan   
3                4  Afghanistan   
4                4  Afghanistan   

                                             Élément           Produit  \
0                            Importations - quantité  Céréales, autres   
1                           Disponibilité intérieure  Céréales, autres   
2                              Aliments pour animaux  Céréales, autres   
3                                         Nourriture  Céréales, autres   
4  Disponibilité alimentaire en quantité (kg/pers...  Céréales, autres   

           Unité  Valeur  
0         1000 t   30.00  
1         1000 t   30.00  
2         1000 t    7.00  
3         1000 t   23.00  
4  kcal/personne    0.57

In [22]:
#Valeur unique Élément
valeurs_uniques = df_céréale['Élément'].unique()
print("Valeurs uniques dans la colonne 'Élément' :", valeurs_uniques)

Valeurs uniques dans la colonne 'Élément' : ['Importations - quantité' 'Disponibilité intérieure'
 'Aliments pour animaux' 'Nourriture'
 'Disponibilité alimentaire en quantité (kg/personne/an)' 'Production'
 'Semences']


In [23]:
#Filtre Élément
df_céréale= df_céréale[df_céréale['Élément'] == 'Aliments pour animaux']
df_céréale.head(5)

Code zone (M49)            Zone                Élément           Produit  \
2                 4     Afghanistan  Aliments pour animaux  Céréales, autres   
19              710  Afrique du Sud  Aliments pour animaux  Céréales, autres   
36                8         Albanie  Aliments pour animaux  Céréales, autres   
53               12         Algérie  Aliments pour animaux  Céréales, autres   
70              276       Allemagne  Aliments pour animaux  Céréales, autres   

     Unité  Valeur  
2   1000 t     7.0  
19  1000 t   105.0  
36  1000 t     0.0  
53  1000 t     9.0  
70  1000 t  1565.0

In [24]:
#Renomme la colonne Valeur
df_céréale = df_céréale.rename(columns={"Valeur": "Céréales_pour_animaux(1000t)"})
df_céréale.head(5)

Code zone (M49)            Zone                Élément           Produit  \
2                 4     Afghanistan  Aliments pour animaux  Céréales, autres   
19              710  Afrique du Sud  Aliments pour animaux  Céréales, autres   
36                8         Albanie  Aliments pour animaux  Céréales, autres   
53               12         Algérie  Aliments pour animaux  Céréales, autres   
70              276       Allemagne  Aliments pour animaux  Céréales, autres   

     Unité  Céréales_pour_animaux(1000t)  
2   1000 t                           7.0  
19  1000 t                         105.0  
36  1000 t                           0.0  
53  1000 t                           9.0  
70  1000 t                        1565.0

In [25]:
#Suprression des colonnes inutiles
df_céréale.drop(columns=['Élément', 'Produit','Unité'], inplace=True)
df_céréale.head(5)

Code zone (M49)            Zone  Céréales_pour_animaux(1000t)
2                 4     Afghanistan                           7.0
19              710  Afrique du Sud                         105.0
36                8         Albanie                           0.0
53               12         Algérie                           9.0
70              276       Allemagne                        1565.0

In [26]:
#Récupération information Viande de Volailles
df_volailles = df_disp[df_disp['Produit'].str.contains('Viande de Volailles', case=False, na=False)]
df_volailles.head(5)

Code zone (M49)         Zone  \
5                4  Afghanistan   
6                4  Afghanistan   
7                4  Afghanistan   
8                4  Afghanistan   
9                4  Afghanistan   

                                             Élément              Produit  \
5                                         Production  Viande de Volailles   
6                            Importations - quantité  Viande de Volailles   
7                           Disponibilité intérieure  Viande de Volailles   
8                                         Nourriture  Viande de Volailles   
9  Disponibilité alimentaire en quantité (kg/pers...  Viande de Volailles   

           Unité  Valeur  
5         1000 t   29.00  
6         1000 t    7.00  
7         1000 t   36.00  
8         1000 t   36.00  
9  kcal/personne    0.87

In [27]:
#Valeur unique Élément
valeurs_uniques = df_volailles['Élément'].unique()
print("Valeurs uniques dans la colonne 'Élément' :", valeurs_uniques)

Valeurs uniques dans la colonne 'Élément' : ['Production' 'Importations - quantité' 'Disponibilité intérieure'
 'Nourriture' 'Disponibilité alimentaire en quantité (kg/personne/an)']


In [28]:
#Filtre d'Element
df_volailles = df_volailles[df_volailles['Élément'].isin(['Production', 'Importations - quantité'])]
df_volailles.head(5)

Code zone (M49)            Zone                  Élément  \
5                 4     Afghanistan               Production   
6                 4     Afghanistan  Importations - quantité   
23              710  Afrique du Sud               Production   
24              710  Afrique du Sud  Importations - quantité   
39                8         Albanie               Production   

                Produit   Unité  Valeur  
5   Viande de Volailles  1000 t    29.0  
6   Viande de Volailles  1000 t     7.0  
23  Viande de Volailles  1000 t  1958.0  
24  Viande de Volailles  1000 t   304.0  
39  Viande de Volailles  1000 t    18.0

In [29]:
df_volailles = df_volailles.pivot(index=[col for col in df_volailles.columns if col not in ['Élément', 'Valeur']], 
                                  columns='Élément', 
                                  values='Valeur').reset_index()
df_volailles.head(5)

Élément  Code zone (M49)                Zone              Produit   Unité  \
0                      4         Afghanistan  Viande de Volailles  1000 t   
1                      8             Albanie  Viande de Volailles  1000 t   
2                     12             Algérie  Viande de Volailles  1000 t   
3                     24              Angola  Viande de Volailles  1000 t   
4                     28  Antigua-et-Barbuda  Viande de Volailles  1000 t   

Élément  Importations - quantité  Production  
0                            7.0        29.0  
1                           47.0        18.0  
2                            0.0       275.0  
3                          316.0        59.0  
4                            7.0         0.0

In [30]:
#Renomme les colonnes
df_volailles = df_volailles.rename(columns={"Importations - quantité": "Import_volailles(1000t)"})
df_volailles = df_volailles.rename(columns={"Production": "Prod_volailles(1000t)"})
df_volailles.head(5)

Élément  Code zone (M49)                Zone              Produit   Unité  \
0                      4         Afghanistan  Viande de Volailles  1000 t   
1                      8             Albanie  Viande de Volailles  1000 t   
2                     12             Algérie  Viande de Volailles  1000 t   
3                     24              Angola  Viande de Volailles  1000 t   
4                     28  Antigua-et-Barbuda  Viande de Volailles  1000 t   

Élément  Import_volailles(1000t)  Prod_volailles(1000t)  
0                            7.0                   29.0  
1                           47.0                   18.0  
2                            0.0                  275.0  
3                          316.0                   59.0  
4                            7.0                    0.0

In [31]:
#Suprression des colonnes inutiles
df_volailles.drop(columns=['Produit', 'Unité'], inplace=True)
df_volailles.head(5)

Élément  Code zone (M49)                Zone  Import_volailles(1000t)  \
0                      4         Afghanistan                      7.0   
1                      8             Albanie                     47.0   
2                     12             Algérie                      0.0   
3                     24              Angola                    316.0   
4                     28  Antigua-et-Barbuda                      7.0   

Élément  Prod_volailles(1000t)  
0                         29.0  
1                         18.0  
2                        275.0  
3                         59.0  
4                          0.0

**2.4 Analyse du fichier Fibl_Organic_2023.CSV**

In [33]:
df_org.head(5)

Unnamed: 0 Unnamed: 1     Item                    No details  \
0      Country       Year  Element  Organic area (farmland) [ha]   
1  Afghanistan       2023      NaN                            28   
2      Albanie       2023      NaN                        736,38   
3      Algérie       2023      NaN                        3349,5   
4      Andorre       2023      NaN                          2,11   

                                 Unnamed: 4         Unnamed: 5  \
0  Organic area share of total farmland [%]  Organic producers   
1                                       NaN                NaN   
2                                      0,06                150   
3                                      0,01                  6   
4                                      0,01                NaN   

                         Unnamed: 6  
0  Organic retail sales [Million €]  
1                               NaN  
2                               NaN  
3                               NaN  
4                               NaN

In [34]:
# Définir la première ligne comme nouvelle en-tête
df_org.columns = df_org.iloc[0] 
df_org = df_org.drop(index=0).reset_index(drop=True)
df_org.head()

0      Country  Year Element Organic area (farmland) [ha]  \
0  Afghanistan  2023     NaN                           28   
1      Albanie  2023     NaN                       736,38   
2      Algérie  2023     NaN                       3349,5   
3      Andorre  2023     NaN                         2,11   
4    Argentine  2023     NaN                  4048'202.80   

0 Organic area share of total farmland [%] Organic producers  \
0                                      NaN               NaN   
1                                     0,06               150   
2                                     0,01                 6   
3                                     0,01               NaN   
4                                     3,43              1368   

0 Organic retail sales [Million €]  
0                              NaN  
1                              NaN  
2                              NaN  
3                              NaN  
4                              NaN

In [35]:
#Renommer les colonnes
df_org.rename(columns={df_org.columns[0]: "Zone"}, inplace=True)
df_org.head()

0         Zone  Year Element Organic area (farmland) [ha]  \
0  Afghanistan  2023     NaN                           28   
1      Albanie  2023     NaN                       736,38   
2      Algérie  2023     NaN                       3349,5   
3      Andorre  2023     NaN                         2,11   
4    Argentine  2023     NaN                  4048'202.80   

0 Organic area share of total farmland [%] Organic producers  \
0                                      NaN               NaN   
1                                     0,06               150   
2                                     0,01                 6   
3                                     0,01               NaN   
4                                     3,43              1368   

0 Organic retail sales [Million €]  
0                              NaN  
1                              NaN  
2                              NaN  
3                              NaN  
4                              NaN

In [36]:
#Suprression des colonnes inutiles
df_org.drop(columns=['Year','Element', 'Organic area (farmland) [ha]','Organic producers', 'Organic retail sales [Million €]'], inplace=True)
df_org.head(5)

0         Zone Organic area share of total farmland [%]
0  Afghanistan                                      NaN
1      Albanie                                     0,06
2      Algérie                                     0,01
3      Andorre                                     0,01
4    Argentine                                     3,43

In [37]:
#Le nombre de valeurs présentes dans chacune des colonnes
print("Nombre de Valeurs présentes dans chacune des colonnes")
print(df_org.count())

Nombre de Valeurs présentes dans chacune des colonnes
0
Zone                                        171
Organic area share of total farmland [%]    159
dtype: int64


In [38]:
#modification type
df_org['Organic area share of total farmland [%]'] = df_org['Organic area share of total farmland [%]'].str.replace(',', '.').astype(float)
df_org['Organic area share of total farmland [%]'] = df_org['Organic area share of total farmland [%]'].astype(float)
print(df_org.dtypes)

0
Zone                                         object
Organic area share of total farmland [%]    float64
dtype: object


**2.5 Analyse du fichier FAOSTAT_population_2023**

In [40]:
#Afficher les dimensions du fichier
print("Le tableau comporte {} observation(s) ou article(s)".format(df_pop.shape[0]))
print("Le tableau comporte {} colonne(s)".format(df_pop.shape[1]))

#La nature des données dans chacune des colonnes
print(df_pop.dtypes)

Le tableau comporte 468 observation(s) ou article(s)
Le tableau comporte 15 colonne(s)
Code Domaine               object
Domaine                    object
Code zone (M49)             int64
Zone                       object
Code Élément                int64
Élément                    object
Code Produit                int64
Produit                    object
Code année                  int64
Année                       int64
Unité                      object
Valeur                    float64
Symbole                    object
Description du Symbole     object
Note                      float64
dtype: object


In [41]:
##Affichage les 5 premières lignes de la table
df_pop.head(5)

Code Domaine                       Domaine  Code zone (M49)            Zone  \
0           OA  Séries temporelles annuelles                4     Afghanistan   
1           OA  Séries temporelles annuelles                4     Afghanistan   
2           OA  Séries temporelles annuelles              710  Afrique du Sud   
3           OA  Séries temporelles annuelles              710  Afrique du Sud   
4           OA  Séries temporelles annuelles                8         Albanie   

   Code Élément            Élément  Code Produit                 Produit  \
0           511  Population totale          3010  Population-Estimations   
1           511  Population totale          3010  Population-Estimations   
2           511  Population totale          3010  Population-Estimations   
3           511  Population totale          3010  Population-Estimations   
4           511  Population totale          3010  Population-Estimations   

   Code année  Année    Unité     Valeur Symbole  \
0        2018   2018  1000 No  36743.039       X   
1        2023   2023  1000 No  41454.761       X   
2        2018   2018  1000 No  58613.000       X   
3        2023   2023  1000 No  63212.384       X   
4        2018   2018  1000 No   2894.231       X   

              Description du Symbole  Note  
0  Ciffre de sources internationales   NaN  
1  Ciffre de sources internationales   NaN  
2  Ciffre de sources internationales   NaN  
3  Ciffre de sources internationales   NaN  
4  Ciffre de sources internationales   NaN

In [42]:
# Garder uniquement les colonnes spécifiées
df_pop = df_pop[['Code zone (M49)','Zone', 'Année', 'Valeur']]

# Afficher le résultat
df_pop.head()

Code zone (M49)            Zone  Année     Valeur
0                4     Afghanistan   2018  36743.039
1                4     Afghanistan   2023  41454.761
2              710  Afrique du Sud   2018  58613.000
3              710  Afrique du Sud   2023  63212.384
4                8         Albanie   2018   2894.231

In [43]:
# Calcul de la croissance de la population par pays
df_pivot = df_pop.pivot(index='Zone', columns='Année', values='Valeur')

# Calcul de la croissance de la population (croissance = (2023 - 2018) / 2018 * 100)
df_pivot['Croissance_population(2018-2023)'] = ((df_pivot[2023] - df_pivot[2018]) / df_pivot[2018]) * 100

# Affichage du résultat avec la croissance
print("\nRésultats avec la croissance par pays:")
df_pivot[['Croissance_population(2018-2023)']]


Résultats avec la croissance par pays:


Année                      Croissance_population(2018-2023)
Zone                                                       
Afghanistan                                       12.823441
Afrique du Sud                                     7.847037
Albanie                                           -2.853124
Algérie                                            8.608825
Allemagne                                          1.415074
...                                                     ...
Îles Salomon                                      12.819612
Îles Turques-et-Caïques                           11.162444
Îles Vierges américaines                          -7.151524
Îles Vierges britanniques                         12.488098
Îles Wallis-et-Futuna                             -3.644068

[234 rows x 1 columns]

In [44]:
#Recuperation de la colonne Croissance
df_pop = pd.merge(df_pop, df_pivot[['Croissance_population(2018-2023)']], on='Zone', how='left')
df_pop.head()

Code zone (M49)            Zone  Année     Valeur  \
0                4     Afghanistan   2018  36743.039   
1                4     Afghanistan   2023  41454.761   
2              710  Afrique du Sud   2018  58613.000   
3              710  Afrique du Sud   2023  63212.384   
4                8         Albanie   2018   2894.231   

   Croissance_population(2018-2023)  
0                         12.823441  
1                         12.823441  
2                          7.847037  
3                          7.847037  
4                         -2.853124

In [45]:
# Filtrer uniquement les données pour l'année 2023
df_pop = df_pop[df_pop['Année'] == 2023]
df_pop.head()

Code zone (M49)            Zone  Année     Valeur  \
1                4     Afghanistan   2023  41454.761   
3              710  Afrique du Sud   2023  63212.384   
5                8         Albanie   2023   2811.655   
7               12         Algérie   2023  46164.219   
9              276       Allemagne   2023  84548.231   

   Croissance_population(2018-2023)  
1                         12.823441  
3                          7.847037  
5                         -2.853124  
7                          8.608825  
9                          1.415074

In [46]:
#Renommer les colonnes et supression
df_pop.rename(columns={df_pop.columns[3]: "Population(1000)"}, inplace=True)
df_pop.drop(columns=['Année'], inplace=True)
df_pop.head()

Code zone (M49)            Zone  Population(1000)  \
1                4     Afghanistan         41454.761   
3              710  Afrique du Sud         63212.384   
5                8         Albanie          2811.655   
7               12         Algérie         46164.219   
9              276       Allemagne         84548.231   

   Croissance_population(2018-2023)  
1                         12.823441  
3                          7.847037  
5                         -2.853124  
7                          8.608825  
9                          1.415074

**2.6 Analyse du fichier FAOSTAT_investissement_2022**

In [48]:
#Afficher les dimensions du fichier
print("Le tableau comporte {} observation(s) ou article(s)".format(df_inv.shape[0]))
print("Le tableau comporte {} colonne(s)".format(df_inv.shape[1]))

#La nature des données dans chacune des colonnes
print(df_inv.dtypes)

Le tableau comporte 193 observation(s) ou article(s)
Le tableau comporte 15 colonne(s)
Code Domaine               object
Domaine                    object
Code zone (M49)             int64
Zone                       object
Code Élément                int64
Élément                    object
Code Produit                int64
Produit                    object
Code année                  int64
Année                       int64
Unité                      object
Valeur                    float64
Symbole                    object
Description du Symbole     object
Note                      float64
dtype: object


In [49]:
#Affichage les 5 premières lignes de la table
df_inv.head(5)

Code Domaine                                           Domaine  \
0         CISP  Profil statistique national sur l’investissement   
1         CISP  Profil statistique national sur l’investissement   
2         CISP  Profil statistique national sur l’investissement   
3         CISP  Profil statistique national sur l’investissement   
4         CISP  Profil statistique national sur l’investissement   

   Code zone (M49)            Zone  Code Élément                  Élément  \
0                4     Afghanistan          6129  Croissance annuelle US$   
1              710  Afrique du Sud          6129  Croissance annuelle US$   
2                8         Albanie          6129  Croissance annuelle US$   
3               12         Algérie          6129  Croissance annuelle US$   
4              276       Allemagne          6129  Croissance annuelle US$   

   Code Produit                                            Produit  \
0         22030  Formation brute de capital fixe (Agriculture, ...   
1         22030  Formation brute de capital fixe (Agriculture, ...   
2         22030  Formation brute de capital fixe (Agriculture, ...   
3         22030  Formation brute de capital fixe (Agriculture, ...   
4         22030  Formation brute de capital fixe (Agriculture, ...   

   Code année  Année Unité    Valeur Symbole Description du Symbole  Note  
0        2022   2022     % -1.509673       E         Valeur estimée   NaN  
1        2022   2022     %  3.300204       E         Valeur estimée   NaN  
2        2022   2022     % -3.049017       E         Valeur estimée   NaN  
3        2022   2022     %  9.106992       E         Valeur estimée   NaN  
4        2022   2022     %  0.532028       E         Valeur estimée   NaN

In [50]:
# Garder uniquement les colonnes spécifiées
df_inv = df_inv[['Code zone (M49)',	'Zone', 'Valeur']]

# Afficher le résultat
df_inv.head()

Code zone (M49)            Zone    Valeur
0                4     Afghanistan -1.509673
1              710  Afrique du Sud  3.300204
2                8         Albanie -3.049017
3               12         Algérie  9.106992
4              276       Allemagne  0.532028

In [51]:
#Renommer les colonnes 
df_inv = df_inv.rename(columns={"Valeur": "Croissance_Investissement(%)"})
df_inv.head()

Code zone (M49)            Zone  Croissance_Investissement(%)
0                4     Afghanistan                     -1.509673
1              710  Afrique du Sud                      3.300204
2                8         Albanie                     -3.049017
3               12         Algérie                      9.106992
4              276       Allemagne                      0.532028

**3 jointure des dataframe**

In [53]:
# Liste de tes DataFrames FAO
dfs = [df_pop,df_pol,df_pib,df_inv,df_céréale,df_volailles]  

# Fusion progressive sur 'code Zone' des df_FAO
df_merge = reduce(lambda left, right: pd.merge(left, right, on='Code zone (M49)', how='outer', suffixes=('', '_dup')), dfs)

# Supprimer les colonnes doublons qui finissent par "_dup"
df_merge = df_merge.loc[:, ~df_merge.columns.str.endswith('_dup')]

df_merge.head(5)

Code zone (M49)               Zone  Population(1000)  \
0                4        Afghanistan         41454.761   
1                8            Albanie          2811.655   
2               12            Algérie         46164.219   
3               16  Samoa américaines            47.521   
4               20            Andorre            80.856   

   Croissance_population(2018-2023)  Stabilité politique  PIB(Int$/cap)  \
0                         12.823441                -2.55            NaN   
1                         -2.853124                 0.11        17261.0   
2                          8.608825                -0.74        13090.9   
3                         -6.653178                 1.13            NaN   
4                          7.575637                 1.59        63379.0   

   Croissance_Investissement(%)  Céréales_pour_animaux(1000t)  \
0                     -1.509673                           7.0   
1                     -3.049017                           0.0   
2                      9.106992                           9.0   
3                           NaN                           NaN   
4                     -5.745133                           NaN   

   Import_volailles(1000t)  Prod_volailles(1000t)  
0                      7.0                   29.0  
1                     47.0                   18.0  
2                      0.0                  275.0  
3                      NaN                    NaN  
4                      NaN                    NaN

In [54]:
#Le nombre de valeurs présentes dans chacune des colonnes
print("Nombre de Valeurs présentes dans chacune des colonnes")
print(df_merge.count())

Nombre de Valeurs présentes dans chacune des colonnes
Code zone (M49)                     234
Zone                                234
Population(1000)                    234
Croissance_population(2018-2023)    234
Stabilité politique                 196
PIB(Int$/cap)                       183
Croissance_Investissement(%)        193
Céréales_pour_animaux(1000t)        178
Import_volailles(1000t)             189
Prod_volailles(1000t)               186
dtype: int64


In [55]:
# Liste unique des pays
pays_unique = list(set(df_merge["Zone"]).union(set(df_org["Zone"])))

# Créer un dictionnaire de correspondance automatique
correspondance = {p: process.extractOne(p, pays_unique)[0] for p in pays_unique}

# Appliquer cette correspondance
df_merge['Zone'] = df_merge['Zone'].replace(correspondance)
df_org['Zone'] = df_org['Zone'].replace(correspondance)

In [56]:
# Convertir en sets
zones_df_merge = set(df_merge['Zone'].dropna().unique())
zones_df_org = set(df_org['Zone'].dropna().unique())

# Valeurs communes
zones_communes = zones_df_merge.intersection(zones_df_org)
print(f"Zones communes ({len(zones_communes)}):\n", zones_communes)

# Valeurs uniques à df1
zones_uniques_df_merge = zones_df_merge - zones_df_org
print(f"\nzones_uniques_df_merge ({len(zones_uniques_df_merge)}):\n", zones_uniques_df_merge)

# Valeurs uniques à df2
zones_uniques_df_org = zones_df_org - zones_df_merge
print(f"\nzones_uniques_df_org ({len(zones_uniques_df_org)}):\n", zones_uniques_df_org)


Zones communes (147):
 {'Bosnie-Herzégovine', 'Bangladesh', 'Népal', 'Islande', 'Soudan', 'Afghanistan', 'Slovénie', 'Croatie', 'Bénin', 'Espagne', 'Zambie', 'Cameroun', 'Seychelles', 'Lituanie', 'Rwanda', 'Namibie', 'Philippines', 'Thaïlande', 'Émirats arabes unis', 'Koweït', 'Mayotte', 'Suisse', 'Arménie', 'Kazakhstan', 'Canada', 'Bahreïn', 'Italie', 'Afrique du Sud', 'Congo', 'Finlande', 'Madagascar', 'Malawi', 'Turquie', 'Équateur', 'Danemark', 'Eswatini', 'Géorgie', 'Îles Salomon', 'République démocratique du Congo', 'Tadjikistan', 'Sierra Leone', 'Chili', 'Kenya', 'Israël', 'Allemagne', 'Brésil', 'Niger', 'Guadeloupe', 'Nouvelle-Zélande', 'Sri Lanka', 'Îles Cook', 'Samoa', 'Bahamas', 'Maroc', 'Indonésie', 'Égypte', 'Autriche', 'Dominique', 'Grèce', 'Ukraine', 'Zimbabwe', 'Suède', 'Guatemala', 'Pérou', 'Australie', 'Chine', 'Costa Rica', 'Andorre', 'Jordanie', 'Vanuatu', 'Uruguay', 'Mongolie', 'Serbie', 'Suriname', 'Sénégal', 'Myanmar', 'Fidji', 'Comores', "Côte d'Ivoire", 'Ghana'

In [57]:
correspondance_pays = {
    "République démocratique populaire lao": "Laos",
    "Iran (République islamique d')": "Iran'",
    "États-Unis d'Amérique": "États-Unis",
    "République-Unie de Tanzanie": "Tanzanie",
    "Bélarus": "Biélorussie",
    "Royaume-Uni de Grande-Bretagne et d'Irlande du Nord": "Royaume-Uni",
    "Pays-Bas (Royaume des)": "Pays-Bas",
    "Tchéquie" : "République tchèque",
    "El Salvador" : "Salvador",
    "Venezuela (République bolivarienne du)" : "Venezuela",
    "Réunion" : "La Réunion",
    "Bolivie (État plurinational de)" : "Bolivie",
    "République de Corée" : "Corée du Sud",
    "Maurice" : "Île Maurice",
    "Fédération de Russie" : "Russie",
    "Viet Nam" : "Viêt Nam",
    "Chine, Taiwan Province de'" : "Taïwan",
    "Iraq" : "Irak",
    "Timor-Leste" : "Timor oriental",
    "Nigéria" : "Nigeria",
    "République de Moldova" : "Moldavie"
}

# Appliquer la correspondance
df_merge['Zone'] = df_merge['Zone'].replace(correspondance_pays)
df_org['Zone'] = df_org['Zone'].replace(correspondance_pays)

# Fusionner
df_merge = pd.merge(df_merge, df_org, on="Zone", how="inner")

print(df_merge.count())

Code zone (M49)                             166
Zone                                        166
Population(1000)                            166
Croissance_population(2018-2023)            166
Stabilité politique                         156
PIB(Int$/cap)                               151
Croissance_Investissement(%)                154
Céréales_pour_animaux(1000t)                150
Import_volailles(1000t)                     155
Prod_volailles(1000t)                       155
Organic area share of total farmland [%]    154
dtype: int64


In [58]:
#Filtre Nan sur céréales
céréale_nan = df_merge[df_merge["Céréales_pour_animaux(1000t)"].isna()]
céréale_nan

Code zone (M49)                       Zone  Population(1000)  \
3                 20                    Andorre            80.856   
32               175                    Mayotte           316.015   
35               184                  Îles Cook            14.222   
49               234                 Îles Féroé            54.714   
53               254           Guyane française           303.402   
56               275                  Palestine          5409.202   
61               312                 Guadeloupe           376.517   
87               438              Liechtenstein            39.598   
95               474                 Martinique           346.002   
107              548                    Vanuatu           320.409   
115              598  Papouasie-Nouvelle-Guinée         10389.635   
122              638                 La Réunion           874.883   
133              702                  Singapour          5789.090   
141              729                     Soudan         50042.791   
149              776                      Tonga           104.597   
165              894                     Zambie         20723.965   

     Croissance_population(2018-2023)  Stabilité politique  PIB(Int$/cap)  \
3                            7.575637                 1.59        63379.0   
32                          17.177129                  NaN            NaN   
35                         -14.304652                  NaN            NaN   
49                           7.210879                  NaN            NaN   
53                           7.986845                  NaN            NaN   
56                          11.516326                -1.82         5744.8   
61                          -9.312346                  NaN            NaN   
87                           3.852711                  NaN            NaN   
95                          -4.967741                  NaN            NaN   
107                         12.322135                 0.95         3074.1   
115                         10.592587                -0.61         4141.2   
122                          2.093607                  NaN            NaN   
133                          2.746022                 1.46       129084.0   
141                         13.140666                -2.00         3299.9   
149                         -0.918848                 1.11            NaN   
165                         15.302448                 0.10         3610.7   

     Croissance_Investissement(%)  Céréales_pour_animaux(1000t)  \
3                       -5.745133                           NaN   
32                            NaN                           NaN   
35                            NaN                           NaN   
49                            NaN                           NaN   
53                            NaN                           NaN   
56                       2.571910                           NaN   
61                            NaN                           NaN   
87                            NaN                           NaN   
95                            NaN                           NaN   
107                      5.105996                           NaN   
115                     10.726990                           NaN   
122                           NaN                           NaN   
133                      4.654887                           NaN   
141                           NaN                           NaN   
149                      3.635366                           NaN   
165                     25.651050                           NaN   

     Import_volailles(1000t)  Prod_volailles(1000t)  \
3                        NaN                    NaN   
32                       NaN                    NaN   
35                       NaN                    NaN   
49                       NaN                    NaN   
53                       NaN                    NaN   
56                       NaN                    NaN

In [59]:
#Remplacement des valeurs céréales par 0
df_merge["Céréales_pour_animaux(1000t)"] = df_merge["Céréales_pour_animaux(1000t)"].fillna(0)
print(df_merge.count())

Code zone (M49)                             166
Zone                                        166
Population(1000)                            166
Croissance_population(2018-2023)            166
Stabilité politique                         156
PIB(Int$/cap)                               151
Croissance_Investissement(%)                154
Céréales_pour_animaux(1000t)                166
Import_volailles(1000t)                     155
Prod_volailles(1000t)                       155
Organic area share of total farmland [%]    154
dtype: int64


In [60]:
#Filtre Nan sur Stabilité politique 
politique_nan = df_merge[df_merge["Stabilité politique"].isna()]
politique_nan

Code zone (M49)                 Zone  Population(1000)  \
29               159                Chine       1454058.724   
32               175              Mayotte           316.015   
35               184            Îles Cook            14.222   
49               234           Îles Féroé            54.714   
53               254     Guyane française           303.402   
54               258  Polynésie française           281.118   
61               312           Guadeloupe           376.517   
87               438        Liechtenstein            39.598   
95               474           Martinique           346.002   
122              638           La Réunion           874.883   

     Croissance_population(2018-2023)  Stabilité politique  PIB(Int$/cap)  \
29                           0.224738                  NaN        21262.3   
32                          17.177129                  NaN            NaN   
35                         -14.304652                  NaN            NaN   
49                           7.210879                  NaN            NaN   
53                           7.986845                  NaN            NaN   
54                           0.465665                  NaN            NaN   
61                          -9.312346                  NaN            NaN   
87                           3.852711                  NaN            NaN   
95                          -4.967741                  NaN            NaN   
122                          2.093607                  NaN            NaN   

     Croissance_Investissement(%)  Céréales_pour_animaux(1000t)  \
29                       2.303339                        3381.0   
32                            NaN                           0.0   
35                            NaN                           0.0   
49                            NaN                           0.0   
53                            NaN                           0.0   
54                      -4.823170                           1.0   
61                            NaN                           0.0   
87                            NaN                           0.0   
95                            NaN                           0.0   
122                           NaN                           0.0   

     Import_volailles(1000t)  Prod_volailles(1000t)  \
29                    1871.0                24194.0   
32                       NaN                    NaN   
35                       NaN                    NaN   
49                       NaN                    NaN   
53                       NaN                    NaN   
54                      18.0                    1.0   
61                       NaN                    NaN   
87                       NaN                    NaN   
95                       NaN                    NaN   
122                      NaN                    NaN   

     Organic area share of total farmland [%]  
29                                       0.66  
32                                       0.85  
35                                       0.48  
49                                       8.37  
53                                      12.96  
54                                       9.01  
61                                       2.62  
87                                      44.58  
95                                       3.02  
122                                      5.03

In [61]:
# Remplacement de la valeur de la Chine
df_merge.loc[df_merge["Zone"] == "Chine", "Stabilité politique"] = -1

# Suppression des lignes où la colonne 'politique' est NaN, sauf celles où 'Zone' est "Chine"
df_merge = df_merge[~((df_merge["Stabilité politique"].isna()) & (df_merge["Zone"] != "Chine"))]

# Affichage du comptage des lignes après modifications
print(df_merge.count())


Code zone (M49)                             157
Zone                                        157
Population(1000)                            157
Croissance_population(2018-2023)            157
Stabilité politique                         157
PIB(Int$/cap)                               151
Croissance_Investissement(%)                153
Céréales_pour_animaux(1000t)                157
Import_volailles(1000t)                     154
Prod_volailles(1000t)                       154
Organic area share of total farmland [%]    145
dtype: int64


In [62]:
#Filtre Nan sur PIB(Int$/cap)
pib_nan = df_merge[df_merge["PIB(Int$/cap)"].isna()]
pib_nan

Code zone (M49)         Zone  Population(1000)  \
0                  4  Afghanistan         41454.761   
13                64      Bhoutan           786.385   
38               192         Cuba         11019.931   
83               422        Liban          5773.493   
149              776        Tonga           104.597   
163              862    Venezuela         28300.854   

     Croissance_population(2018-2023)  Stabilité politique  PIB(Int$/cap)  \
0                           12.823441                -2.55            NaN   
13                           3.504532                 0.86            NaN   
38                          -1.845416                 0.50            NaN   
83                          -3.519750                -1.47            NaN   
149                         -0.918848                 1.11            NaN   
163                         -5.054880                -1.26            NaN   

     Croissance_Investissement(%)  Céréales_pour_animaux(1000t)  \
0                       -1.509673                           7.0   
13                       4.810530                           0.0   
38                       5.926995                           0.0   
83                      28.952468                           6.0   
149                      3.635366                           0.0   
163                           NaN                           0.0   

     Import_volailles(1000t)  Prod_volailles(1000t)  \
0                        7.0                   29.0   
13                       3.0                    1.0   
38                     317.0                   19.0   
83                      12.0                  120.0   
149                      9.0                    0.0   
163                     23.0                  426.0   

     Organic area share of total farmland [%]  
0                                         NaN  
13                                       0.72  
38                                       0.08  
83                                       0.21  
149                                       NaN  
163                                      0.01

In [63]:
# Récupération des valeurs PIB précedent 2021
df_merge.loc[df_merge["Zone"] == "Afghanistan", "PIB(Int$/cap)"] =  2138.9
df_merge.loc[df_merge["Zone"] == "Bhoutan", "PIB(Int$/cap)"] = 13423.4
df_merge.loc[df_merge["Zone"] == "Cuba", "PIB(Int$/cap)"] = 12000 #Estimation chatgpt
df_merge.loc[df_merge["Zone"] == "Liban", "PIB(Int$/cap)"] = 11860.1
df_merge.loc[df_merge["Zone"] == "Tonga", "PIB(Int$/cap)"] = 6778.5
# Suppression des lignes où la colonne 'PIB ' est NaN, 
df_merge = df_merge[df_merge["Zone"] != "Venezuela"]
print(df_merge.count())

Code zone (M49)                             156
Zone                                        156
Population(1000)                            156
Croissance_population(2018-2023)            156
Stabilité politique                         156
PIB(Int$/cap)                               156
Croissance_Investissement(%)                153
Céréales_pour_animaux(1000t)                156
Import_volailles(1000t)                     153
Prod_volailles(1000t)                       153
Organic area share of total farmland [%]    144
dtype: int64


In [64]:
#Filtre Nan sur Organic area share of total farmland [%]
org_nan = df_merge[df_merge["Organic area share of total farmland [%]"].isna()]
org_nan

Code zone (M49)         Zone  Population(1000)  \
0                  4  Afghanistan         41454.761   
21               108      Burundi         13689.450   
27               148        Tchad         19319.064   
30               170     Colombie         52321.152   
60               308      Grenade           117.081   
74               388     Jamaïque          2839.786   
106              528     Pays-Bas         18092.524   
110              562        Niger         26159.867   
131              690   Seychelles           127.951   
134              703    Slovaquie          5518.055   
137              706      Somalie         18358.615   
149              776        Tonga           104.597   

     Croissance_population(2018-2023)  Stabilité politique  PIB(Int$/cap)  \
0                           12.823441                -2.55         2138.9   
21                          15.430771                -1.19          857.2   
27                          19.574332                -1.48         1748.5   
30                           6.724575                -0.64        18732.7   
60                           1.201476                 1.04        15136.3   
74                           0.698167                 0.38        10114.0   
106                          3.853012                 0.72        69953.7   
110                         17.900602                -1.47         1652.7   
131                         11.223053                 0.76        23409.6   
134                          1.374606                 0.44        38566.3   
137                         18.806863                -2.48         1452.3   
149                         -0.918848                 1.11         6778.5   

     Croissance_Investissement(%)  Céréales_pour_animaux(1000t)  \
0                       -1.509673                           7.0   
21                      11.373277                           7.0   
27                      -6.832992                          17.0   
30                       1.116099                          12.0   
60                       4.450409                           0.0   
74                       3.949839                           0.0   
106                      1.648044                         207.0   
110                    -19.088605                         105.0   
131                     13.097800                           0.0   
134                      3.318049                          31.0   
137                     11.305287                           1.0   
149                      3.635366                           0.0   

     Import_volailles(1000t)  Prod_volailles(1000t)  \
0                        7.0                   29.0   
21                       0.0                    8.0   
27                       3.0                    7.0   
30                      81.0                 1820.0   
60                       8.0                    0.0   
74                      21.0                  140.0   
106                    897.0                  876.0   
110                      2.0                   20.0   
131                      4.0                    2.0   
134                     86.0                   51.0   
137                     10.0                    4.0   
149                      9.0                    0.0   

     Organic area share of total farmland [%]  
0                                         NaN  
21                                        NaN  
27                                        NaN  
30                                        NaN  
60                                        NaN  
74                                        NaN  
106                                       NaN  
110                                       NaN  
131                                       NaN  
134                                       NaN  
137                                       NaN  
149                                       NaN

In [65]:
#Vérification des valeurs Organic area share of total farmland [%]
print(df_merge['Organic area share of total farmland [%]'].min())
print(df_merge['Organic area share of total farmland [%]'].max())

0.0
27.34


In [66]:
# Estimation des valeurs
df_merge.loc[df_merge["Organic area share of total farmland [%]"] == "Pays-Bas", "Organic area share of total farmland [%]"] =  4.41 #Valeur 2022 
#Remplacement des valeurs Organic area share of total farmland [%] par 0
df_merge["Organic area share of total farmland [%]"] = df_merge["Organic area share of total farmland [%]"].fillna(0)
print(df_merge.count())

Code zone (M49)                             156
Zone                                        156
Population(1000)                            156
Croissance_population(2018-2023)            156
Stabilité politique                         156
PIB(Int$/cap)                               156
Croissance_Investissement(%)                153
Céréales_pour_animaux(1000t)                156
Import_volailles(1000t)                     153
Prod_volailles(1000t)                       153
Organic area share of total farmland [%]    156
dtype: int64


In [67]:
#Filtre Nan sur #Filtre Nan sur Croissance_Investissement(%)
inv_nan = df_merge[df_merge["Croissance_Investissement(%)"].isna()]
inv_nan

Code zone (M49)      Zone  Population(1000)  \
99               498  Moldavie          3067.070   
141              729    Soudan         50042.791   
164              882     Samoa           216.663   

     Croissance_population(2018-2023)  Stabilité politique  PIB(Int$/cap)  \
99                          -2.749183                -0.67        15229.5   
141                         13.140666                -2.00         3299.9   
164                          4.374657                 1.08         5689.0   

     Croissance_Investissement(%)  Céréales_pour_animaux(1000t)  \
99                            NaN                           1.0   
141                           NaN                           0.0   
164                           NaN                           0.0   

     Import_volailles(1000t)  Prod_volailles(1000t)  \
99                      23.0                   46.0   
141                      1.0                   85.0   
164                     17.0                    0.0   

     Organic area share of total farmland [%]  
99                                       1.82  
141                                      0.04  
164                                     14.60

In [68]:
#Remplacement des valeurs Croissance_Investissement(%) par 0
df_merge["Croissance_Investissement(%)"] = df_merge["Croissance_Investissement(%)"].fillna(0)
print(df_merge.count())

Code zone (M49)                             156
Zone                                        156
Population(1000)                            156
Croissance_population(2018-2023)            156
Stabilité politique                         156
PIB(Int$/cap)                               156
Croissance_Investissement(%)                156
Céréales_pour_animaux(1000t)                156
Import_volailles(1000t)                     153
Prod_volailles(1000t)                       153
Organic area share of total farmland [%]    156
dtype: int64


In [69]:
#Filtre Nan sur Import_volailles(1000t)
vol_nan = df_merge[df_merge["Import_volailles(1000t)"].isna()]
vol_nan

Code zone (M49)       Zone  Population(1000)  \
3                 20    Andorre            80.856   
56               275  Palestine          5409.202   
133              702  Singapour          5789.090   

     Croissance_population(2018-2023)  Stabilité politique  PIB(Int$/cap)  \
3                            7.575637                 1.59        63379.0   
56                          11.516326                -1.82         5744.8   
133                          2.746022                 1.46       129084.0   

     Croissance_Investissement(%)  Céréales_pour_animaux(1000t)  \
3                       -5.745133                           0.0   
56                       2.571910                           0.0   
133                      4.654887                           0.0   

     Import_volailles(1000t)  Prod_volailles(1000t)  \
3                        NaN                    NaN   
56                       NaN                    NaN   
133                      NaN                    NaN   

     Organic area share of total farmland [%]  
3                                        0.01  
56                                       1.32  
133                                      2.21

In [70]:
#Remplacement des valeurs Import_volailles(1000t) par 0
df_merge["Import_volailles(1000t)"] = df_merge["Import_volailles(1000t)"].fillna(0)
df_merge["Prod_volailles(1000t)"] = df_merge["Prod_volailles(1000t)"].fillna(0)
print(df_merge.count())

Code zone (M49)                             156
Zone                                        156
Population(1000)                            156
Croissance_population(2018-2023)            156
Stabilité politique                         156
PIB(Int$/cap)                               156
Croissance_Investissement(%)                156
Céréales_pour_animaux(1000t)                156
Import_volailles(1000t)                     156
Prod_volailles(1000t)                       156
Organic area share of total farmland [%]    156
dtype: int64


In [76]:
#Calcul de la population cibler
pop_total_df_pop = df_pop['Population(1000)'].sum()
pop_total_df_merge = df_merge['Population(1000)'].sum()
# Calcul du pourcentage
pourcentage = (pop_total_df_merge / pop_total_df_pop) * 100
# Affichage
print(f"Population totale df_pop : {pop_total_df_pop:,.0f}")
print(f"Population totale df_merge : {pop_total_df_merge:,.0f}")
print(f"Pourcentage représenté par df_merge : {pourcentage:.2f}%")

Population totale df_pop : 9,543,925
Population totale df_merge : 7,762,129
Pourcentage représenté par df_merge : 81.33%


Analyse sur 156 pays et 81.33% de la population

**4 Création du fichier pour analyse**

In [80]:
df_merge.to_csv("df_merge_export.csv", index=False)